# HG changeset patch
# User Michael Godfrey <godfrey@isl.stanford.edu>
# Date 1296592927 18000
#      Tue Feb 01 15:42:07 2011 -0500
# Node ID 11427f71699b86014454ee2e79fc11d27a7240a6
# Parent  6ba28900706b0e3c1e299758f59a8dfe2ae78665
gl2ps-renderer.cc (glps_renderer::draw): return immediately if fdopen fails

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2011-02-01  Michael Godfrey  <godfrey@isl.stanford.edu>
+
+	* gl2ps-renderer.cc (glps_renderer::draw):
+	Return immediately if fdopen fails.
+
 2011-02-01  John W. Eaton  <jwe@octave.org>
 
 	* gl-render.cc:	(opengl_renderer::setup_opengl_transformation,
 	opengl_renderer::draw_axes_planes,
 	opengl_renderer::draw_axes_boxes,
 	opengl_renderer::draw_axes_x_grid,
 	opengl_renderer::draw_axes_y_grid,
 	opengl_renderer::draw_axes_z_grid,
diff --git a/src/gl2ps-renderer.cc b/src/gl2ps-renderer.cc
--- a/src/gl2ps-renderer.cc
+++ b/src/gl2ps-renderer.cc
@@ -39,16 +39,23 @@ glps_renderer::draw (const graphics_obje
 {
   static bool in_draw = false;
 
   if (!in_draw)
     {
       in_draw = true;
 
       FILE *fp = fdopen (fid, "wb");
+
+      if (! fp)
+        {
+          error ("gl2ps-renderer: fdopen failed");
+          return;
+        }
+
       GLint buffsize = 0, state = GL2PS_OVERFLOW;
       GLint viewport[4];
 
       glGetIntegerv (GL_VIEWPORT, viewport);
 
       GLint gl2ps_term;
       if (term.find ("eps") != std::string::npos) gl2ps_term = GL2PS_EPS;
       else if (term.find ("pdf") != std::string::npos) gl2ps_term = GL2PS_PDF;
